src/binary_search_tree.js

```diff
@@ -0,0 +1,192 @@
+class Node {
+  constructor() {
+    this.parent = null
+    this.rightChild = null
+    this.leftChild = null
+    this.data
+  }
+}
+//////HELPER FUNCTIONS
+function hasChildren(node) {
+  return node.rightChild || node.leftChild
+}
+
+function recursiveInsert(startNode,newNode) {
+  if(newNode.data<=startNode.data) {
```

Comment:
from my understanding a new node that is less than the start node goes down the tree through the left and if its greater than it goes down the tree on the right